packages/swapper/src/swappers/RelaySwapper/utils/relayTokenToAsset.ts

```diff
@@ -0,0 +1,31 @@
+import type { AssetId } from '@shapeshiftoss/caip'
+import type { Asset } from '@shapeshiftoss/types'
+
+import { relayTokenToAssetId } from './relayTokenToAssetId'
+import type { RelayToken } from './types'
+
+export const relayTokenToAsset = (
+  relayToken: RelayToken,
+  assets: Partial<Record<AssetId, Asset>>,
+): Asset => {
+  const assetId = relayTokenToAssetId(relayToken)
+  const maybeAsset = assets[assetId]
+  if (maybeAsset) return maybeAsset
+
+  // It shouldn't happen but...
+  // asset not known by shapeshift
+  // create a placeholder asset using the data we have
+  return {
+    assetId,
+    chainId: relayToken.chainId,
```

Comment:
This is incorrect, see https://github.com/shapeshift/web/pull/9188/files#r2027147961

Reply:
Used the generated assetId to get the chainId in [7e3a115](https://github.com/shapeshift/web/pull/9188/commits/7e3a115c7aeb37843ca13844d6f4f73501a40f03)